.github/workflows/omgx-integration.yml

```diff
@@ -0,0 +1,146 @@
+name: omgx_integration
+
+on:
+  push:
+    branches:
+      - 'master'
+      - 'develop'
+      - '*rc'
+      - 'regenesis/*'
+  pull_request:
+  workflow_dispatch:
+
+jobs:
+  omgx_integration:
+    runs-on: ubuntu-latest
+    env:
+        DOCKER_BUILDKIT: 1
+        COMPOSE_DOCKER_CLI_BUILD: 1
+    steps:
+      # Monorepo tests
+      - uses: actions/checkout@v2
+
+      - name: Get yarn cache directory path
+        id: yarn-cache-dir-path
+        run: echo "::set-output name=dir::$(yarn cache dir)"
+
+      - uses: actions/cache@v2
+        id: yarn-cache
+        with:
+          path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
+          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-yarn-
+
+      - name: Install yq
+        run: sudo snap install yq
+
+      - name: Bring the stack up + OMGX
+        working-directory: ./ops
+        env:
+          BUILD: 1
+          DAEMON: 1
+        run: ./up_local.sh
+
+      - name: Background logs
+        working-directory: ./ops
+        run: docker-compose -f docker-compose-omgx.yml -f docker-compose-omgx-services.yml logs --follow &
+
+      - name: Run the integration tests
+        working-directory: ./ops
+        run: docker-compose run integration_tests
+
+      # Examples Tests
+      - name: Test & deploy hardhat-example on Ethereum (regression)
+        working-directory: ./examples/hardhat
+        run: |
+          yarn
+          yarn deploy
+          yarn test:integration
+
+      - name: Test & deploy hardhat-example on Optimistic Ethereum
+        working-directory: ./examples/hardhat
+        run: |
+          yarn deploy:ovm
+          yarn test:integration:ovm
+
+      - name: Test & deploy waffle-example on Ethereum (regression)
+        working-directory: ./examples/waffle
+        run: |
+          yarn
+          yarn compile
+          yarn test:integration
+
+      - name: Test & deploy waffle-example on Optimistic Ethereum
+        working-directory: ./examples/waffle
+        run: |
+          yarn compile:ovm
+          yarn test:integration:ovm
+
+      - name: Test & deploy truffle-example on Ethereum (regression)
+        working-directory: ./examples/truffle
+        run: |
+          yarn
+          yarn compile
+          yarn test:integration
+          yarn deploy
+
+      - name: Test & deploy truffle-example on Optimistic Ethereum
+        working-directory: ./examples/truffle
+        run: |
+          yarn compile:ovm
+          yarn test:integration:ovm
+          yarn deploy:ovm
+
+      - name: Test serving Wallet contracts
```

Comment:
with this we ensure wallet deployer deploys contracts, later we will add wallet integration tests

Reply:
but we need to run all their integration tests so that we make sure our stack does not interfere with the correctness of the optimistic system